lib/spack/spack/solver/asp.py

```diff
@@ -537,6 +561,36 @@ def bootstrap_clingo():
         from clingo import parse_files
 
 
+def stringify(sym):
+    """Stringify symbols from clingo models.
+
+    This will turn a ``clingo.Symbol`` into a string, or a sequence of ``clingo.Symbol``
+    objects into a tuple of strings.
+
+    """
+    # TODO: simplify this when we no longer have to support older clingo versions.
+    if isinstance(sym, (list, tuple)):
```

Comment:
(minor) Here we can probably use:
```suggestion
    if isinstance(sym, collections.abc.Sequence):
```

Reply:
I really hate using this because it's error prone with strings:

```python
>>> isinstance("foo", collections.abc.Sequence)
True
```